L52-objc-bazel-support.md

```diff
@@ -0,0 +1,124 @@
+gRPC Objective-C Bazel Build Support
+----
+* Author: tonyzhehaolu
+* Approver: mxyan
+* Status: Draft
+* Implemented in: Bazel and Starlark
+* Last updated: Today's date
+* Discussion at: ???
+
+## Abstract
+
+Proposes a set of Bazel rules for building iOS applications with gRPC Objective-C library.
+
+
+## Background
+
+The gRPC Objective-C library so far only supports installation via Cocoapods. Requests for Bazel support are continually raised.
+
+In addition to the available native rules (`objc_library` and `proto_library`), `objc_proto_lib` and `objc_grpc_lib` need to be created because the native `objc_proto_library` is actually [not usable](https://github.com/bazelbuild/bazel/issues/7348). Some other rules are also needed in order to compile the actual library.
+
+### Related Proposals
+`objc_proto_lib` and `objc_grpc_lib` are built upon the implementation of the `generate_cc` rule defined in [generate_cc.bzl](https://github.com/grpc/grpc/blob/bazel_test/bazel/generate_cc.bzl). There doesn't seem to be a proposal for that, though.
+
+
+## Proposal
+
+For now, assume that the WORKSPACE root is the gRPC repository.
+
+### Dependency Graph
+
+According to the dependencies in an iOS application that uses gRPC Objc library (shown below), we created two `objc_library` rules in the `src/objective-c` package within the `com_github_grpc_grpc` workspace: `grpc_objc` and `grpc_objc_protoRPC`.
```

Comment:
I tend to name them as `grpc_objc_client` and `proto_objc_rpc` for compatibility with internal targets.